### ISSUE TITLE
[Bug]: Text Positioned Outside Image Not Captured in Editor

### ISSUE THREAD

**Issue by 1041harpreet** [hm21/pro_image_editor#210]
### Package Version

"4.3.6"

### Flutter Version

3.22.1

### Platforms

Android

### How to reproduce?

I used basic example.When we place text over image it works fine but when we moved text outside or below image ,it does not capture.

[Video](https://drive.google.com/file/d/1odjKrCMrsdegGAmMMeghyxeY_6giT6tZ/view?usp=sharing)

### Example code (optional)

```Dart
class EditEventPage extends ConsumerStatefulWidget {
  const EditEventPage({super.key});

  @override
  ConsumerState<EditEventPage> createState() => _EditEventPageState();
}

class _EditEventPageState extends ConsumerState<EditEventPage> {
  final _editorKey = GlobalKey<ProImageEditorState>();

  @override
  Widget build(BuildContext context) {
    return ProImageEditor.network(
      imgUrl,
      key: _editorKey,
      callbacks: ProImageEditorCallbacks(
        onImageEditingComplete: (byte) async {
          try {
            await Navigator.push(
              context,
              PageTransition(
                type: PageTransitionType.bottomToTop,
                duration: const Duration(milliseconds: 500),
                child: AddAudioScreen(
                  imageBytes: byte,
                ),
              ),
            );
          } catch (e) {
            print(e);
          }
        },
      ),
    );
  }
}


//on Add audio screen:

  SizedBox(
                height: 400,
                child: Image.memory(
                  imageBytes,
                  fit: BoxFit.cover,
                ),
              ),
```

### Device Model (optional)

One plus Nord (android 14)

**Comment by hm21** [comment]
By default, the editor captures only the area where the background image is present, excluding everything outside of it. However, if you want to change this behavior to include layers outside that area, you can do so by adding the following configurations to the image editor:

```dart
configs: ProImageEditorConfigs(
    imageGenerationConfigs: const ImageGenerationConfigs(
      captureOnlyBackgroundImageArea: false,

      /// If you want the entire visible area to be captured, rather than only the parts 
      /// containing elements (images or layers), you can set the parameter below to `false`.
      /// captureOnlyDrawingBounds: false,
      
      /// The default output format is `jpg`, which does not support transparency.
      /// To enable transparency, change the format as shown below.
      outputFormat: OutputFormat.png,
    ),
)
```

**Comment by lcundiff** [comment]
@hm21 Commenting here, since my question is directly tied to this answer, but can create a new discussion/issue if needed. 

When I added a transparent background to the initial canvas, it crops to the width of the largest content on the background image instead of the size of the background image due to `captureOnlyDrawingBounds` being true. If the background image is **not** transparent, the exported image is correctly the width of the background image. I need `captureOnlyBackgroundImageArea` to be true, since I want everything outside the background image to be cropped, but I don't want the exported image to be the size of the content when background image is transparent, just the size of the transparent background image.

I believe what I want is for `captureOnlyDrawingBounds` to be false and `captureOnlyBackgroundImageArea` to be true, since I still want everything outside of canvas to be cropped. However this gives an error, although I think this assert may actually be intended for when captureOnlyDrawingBounds is true and captureOnlyBackgroundImageArea is false, not the other way around based of the error message I get from the assert. 

<img width="359" alt="image" src="https://github.com/user-attachments/assets/0c1da33f-7459-42a2-ab6c-4e84afa5a69e">

<img width="952" alt="image" src="https://github.com/user-attachments/assets/7c33b335-b311-48bc-bfbb-b587aa7ad2e3">

**Comment by hm21** [comment]
@lcundiff You are correct, there is an issue with the assert. This assert was necessary in older versions of the editor, but we can now completely remove it, as every combination will function correctly, each with its own benefits. I have just released version `6.0.1`, which resolves the issue.

**Comment by lcundiff** [comment]
Great! Thank you!
